### ISSUE TITLE
Create new Boutiques module 'BoutiquesInputCacheCleaner'

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#1203]
This module would invoke 'cache_erase' on input files, once a task is complete, using a new specification in the descriptor:

```json
"custom": {
  "cbrain:integrator_modules": {
    "BoutiquesInputCacheCleaner": [
      "inputid1",
      "inputid2"
    ]
  }
}
```

That way when processing large datasets in parallel where the dataset is split in chunks (e.g. BidsSubjects of the UKBB), we can erase the subjec't data from the cache and free some disk space.

**Comment by MontrealSergiy** [comment]
@prioux should it support cbcsv lists?

**Comment by prioux** [comment]
cbcsv files are replaced by their individual file components at that point, so there is no need to even implement anything particular.

**Comment by prioux** [comment]
I'm increasing the priority on this one because we'll need it for the UK BioBank processing.

**Comment by prioux** [comment]
I'm adding a requirement. To prevent a task from erasing an input that is also used by another task, make  a check on the timestamp of the SyncStatus object.

After the setup() method, record a timestamp in the meta data of the task:

```ruby
task.meta[:setup_time] = Time.now
```

then just before attempting the cleanup of the input file, fetch its SyncStatus object and compare with the timestamp recorded:

```ruby
setup_time = inputfile.local_sync_status&.accessed_at
if (setup_time - task.meta[:setup_time].abs  < 2.seconds
   erase here
end
```

**Comment by MontrealSergiy** [comment]
1. @prioux I guess I can start with it, but are you sure the timestamp comparison will conflict?- But my feeling this might fail if interleaving (due to other load) tasks running on different processors and different tools can interleave, starting task earlier does not guaranty an early finish.  My filling is a better locking can be devised though with more efforts.

**Comment by MontrealSergiy** [comment]
2. @prioux I tried it with BoutiquesDemo yet it seems that inputfile.local_sync_status is nil.
is it because  local_sync_status is specific to BIDs apps?

**Comment by MontrealSergiy** [comment]
with some effort I managed to get exception (I guess restarting many tasks one after one) 
![image](https://user-images.githubusercontent.com/11949081/157340545-43db74a5-360a-41c3-8389-51817a1d6df3.png)
